pymc3/step_methods/metropolis.py

```diff
@@ -120,9 +120,10 @@ def astep(self, q0):
                 q = (q0 + delta).astype(int)
             else:
                 delta[self.discrete] = round(delta[self.discrete], 0).astype(int)
-                q = q0 + delta
+                q = (q0 + delta)
         else:
-            q = q0 + delta
+            q0 = q0.astype(theano.config.floatX)
+            q = (q0 + delta).astype(theano.config.floatX)
```

Comment:
How do you mean? Theano is always needed and I would expect that `model.dtype` must match `theano.config.floatX`.